test/forkMainnet/LimitOrderSwap/GroupFill.t.sol

```diff
@@ -21,7 +22,19 @@ contract GroupFillTest is LimitOrderSwapTest {
         for (uint256 i = 0; i < makerPrivateKeys.length; ++i) {
             makers[i] = payable(vm.addr(makerPrivateKeys[i]));
             deal(makers[i], 100 ether);
-            setTokenBalanceAndApprove(makers[i], address(limitOrderSwap), tokens, 100000);
+            setTokenBalanceAndApprove(makers[i], UNISWAP_PERMIT2_ADDRESS, tokens, 100000);
+
+            vm.startPrank(makers[i]);
+            for (uint256 j = 0; j < tokens.length; ++j) {
+                // maker should call permit2 first independently
+                IUniswapPermit2(UNISWAP_PERMIT2_ADDRESS).approve(
+                    address(tokens[j]),
+                    address(limitOrderSwap),
+                    type(uint160).max,
+                    uint48(block.timestamp + 1 days)
+                );
+            }
```

Comment:
The another way is using batch permit which required multiple permit sigs. I think approve is simpler.

Reply:
Yeah I was just thinking how to make it easier for maker. Since maker now has to approve twice, one for Permit2 and one for limit order swap. The first one can not be batched but the second one can so that's probably where we should help improve the UX on.

Reply:
I see. Yeah if a maker trying to create multiple orders at the same time, using batch permit can improve UX. But I think there's no need in test?

Reply:
Yeah no need for that in test.